packages/app/src/runner/events/telemetry.ts

```
@@ -0,0 +1,59 @@
+import { telemetry } from '@packages/telemetry/src/browser'
+
+export const addTelemetryListeners = (getCypress) => {
+  const Cypress = getCypress()
+
+  Cypress.on('test:before:run', (attributes, test) => {
+    // we emit the 'test:before:run' events within various driver tests
+    if (test?.fullTitle()) {
+      // If a span for a previous test hasn't been ended, end it before starting the new test span
+      const previousTestSpan = telemetry.findActiveSpan((span) => {
+        return span?.name.startsWith('test:')
+      })
+
+      if (previousTestSpan) {
+        telemetry.endActiveSpanAndChildren(previousTestSpan)
+      }
+
+      const span = telemetry.startSpan({ name: `test:${test.fullTitle()}`, active: true })
+
+      span?.setAttributes({
+        currentRetry: attributes.currentRetry,
+      })
+    }
+  })
+
+  Cypress.on('test:after:run', (attributes, test) => {
+    // I haven't seen a test:after:run event without a test, but just to be safe.
```

We would [fail](https://github.com/cypress-io/cypress/blob/b34897241ea9255a73fed92905d1629ffbfe5540/packages/driver/src/cypress/runner.ts#L48) before we got here if there wasn't a test.

---

this is just to end the span, right? Is there someplace else we can end the span if we fail early where we can maybe refactor this to a method we can call?

---

This is fine as is, I just removed the comment.

---

added a try catch here